### ISSUE TITLE
Column names quoted with backticks (e.g. reserved words) are quoted incorrectly in `Utils.ensureQuoted`

### ISSUE THREAD

**Issue by bclune** [snowflakedb/spark-snowflake#66]
The `Utils.ensureQuoted` and `Utils.isQuoted` functions do not respect identifiers that are already
quoted with backticks and instead wrap the backtick-quoted string in an additional set of double quotes. This presents an issue in particular when specifying reserved-word column
names in the `columnmap` option, as the generated SQL looks like:

```
COPY INTO table_name (..., "`values`", ...)
```

where `values` is the name of a column in the given table.

This causes an error:

```
SQL compilation error: error line 1 at position 102 invalid identifier '"`values`"'
```

**Comment by bclune** [comment]
https://github.com/snowflakedb/spark-snowflake/pull/65

should fix the issue, although I am unable to run integration tests to verify that there is no breakage.

**Comment by bclune** [comment]
From what I can tell, there are two places in the code that use `Utils.ensureQuoted`:

**Comment by binglihub** [comment]
Hi @bclune, this issue was fixed in v2.4.6. Thank you
